external-crates/move/crates/move-binary-format/src/normalized.rs

```diff
@@ -305,337 +662,400 @@ impl Module {
-            .code
-            .as_ref()
-            .map(|code| {
-                code.code
-                    .iter()
-                    .map(|bytecode| Bytecode::new(m, bytecode, &code.jump_tables))
-                    .collect()
-            })
-            .unwrap_or_default();
-        let f = Function {
+        let name = pool.intern(m.identifier_at(fhandle.name));
+        let (jump_tables, code) = if include_code {
+            let jump_tables = def
+                .code
+                .iter()
+                .flat_map(|code| code.jump_tables.iter())
+                .map(|jt| Rc::new(VariantJumpTable::new(tables, jt)))
+                .collect::<Vec<_>>();
+            let code = def
+                .code
+                .as_ref()
+                .map(|code| {
+                    code.code
+                        .iter()
+                        .map(|bytecode| Bytecode::new(tables, pool, m, bytecode, &jump_tables))
+                        .collect()
+                })
+                .unwrap_or_default();
+            (jump_tables, code)
+        } else {
+            (vec![], vec![])
+        };
+        Function {
+            name,
             visibility: def.visibility,
             is_entry: def.is_entry,
             type_parameters: fhandle.type_parameters.clone(),
-            parameters: m
-                .signature_at(fhandle.parameters)
-                .0
-                .iter()
-                .map(|s| Type::new(m, s))
-                .collect(),
-            return_: m
-                .signature_at(fhandle.return_)
-                .0
-                .iter()
-                .map(|s| Type::new(m, s))
-                .collect(),
+            parameters: tables.signatures[fhandle.parameters.0 as usize].clone(),
+            return_: tables.signatures[fhandle.return_.0 as usize].clone(),
+            code_included: include_code,
+            jump_tables,
             code,
-        };
-        (name, f)
+        }
     }
 
-    /// Create a `Function` for function named `func_name` in module `m`.
-    pub fn new_from_name(m: &CompiledModule, func_name: &IdentStr) -> Option<Self> {
-        for func_defs in &m.function_defs {
-            if m.identifier_at(m.function_handle_at(func_defs.function).name) == func_name {
-                return Some(Self::new(m, func_defs).1);
-            }
+    // Panics if `code_included` is `false`.
+    pub fn code(&self) -> &[Bytecode<S>] {
+        assert!(self.code_included);
+        &self.code
+    }
+
+    /// Aborts if `code_included` is `false`.
+    pub fn equals(&self, other: &Self) -> bool
+    where
+        S: Eq,
+    {
+        let Self {
+            name,
+            visibility,
+            is_entry,
+            type_parameters,
+            parameters,
+            return_,
+            code_included,
+            jump_tables,
+            code,
+        } = self;
+        assert!(code_included);
+        assert!(other.code_included);
```

Comment:
What do you think about these being debug asserts and returning `false` if either `code_included` is false?

Reply:
Done!